support/k8s/k8s-workload-registrar/controllers/spiffeid_controller.go

```diff
@@ -0,0 +1,295 @@
+type SpiffeIDReconciler struct {
+	client.Client
+	c                  SpiffeIDReconcilerConfig
+	myId               *string
+	spiffeIDCollection map[string]string
+}
+
+// NewSpiffeIDReconciler creates a new SpiffeIDReconciler object
+func NewSpiffeIDReconciler(config SpiffeIDReconcilerConfig) (*SpiffeIDReconciler, error) {
+	r := &SpiffeIDReconciler {
+		Client: config.Mgr.GetClient(),
+		c: config,
+		spiffeIDCollection:  make(map[string]string),
+	}
+
+	err := ctrl.NewControllerManagedBy(r.c.Mgr).
+		For(&spiffeidv1beta1.SpiffeID{}).
+		Complete(r)
+	if err != nil {
+		return nil, err
+	}
+
+	return r, nil
+}
+
+// Initialize ensures there is a node registration entry for PSAT nodes in the cluster.
+func (r *SpiffeIDReconciler) Initialize(ctx context.Context) error {
+	nodeID := r.nodeID()
+	r.myId = &nodeID
+	return r.createEntry(ctx, &common.RegistrationEntry{
+		ParentId: idutil.ServerID(r.c.TrustDomain),
+		SpiffeId: nodeID,
+		Selectors: []*common.Selector{
+			{Type: "k8s_psat", Value: fmt.Sprintf("cluster:%s", r.c.Cluster)},
+		},
+	})
+}
+
+// Reconcile ensures the SPIRE Server entry matches the corresponding CRD
+func (r *SpiffeIDReconciler) Reconcile(req ctrl.Request) (ctrl.Result, error) {
+	ctx := context.Background()
+	spiffeID := spiffeidv1beta1.SpiffeID{}
+
+	if err := r.Get(ctx, req.NamespacedName, &spiffeID); err != nil {
+		if !k8serrors.IsNotFound(err) {
+			r.c.Log.WithError(err).Error("unable to fetch SpiffeID")
+			return ctrl.Result{}, err
+		}
+
+		// Delete event
+		if err := r.ensureDeleted(ctx, r.spiffeIDCollection[req.NamespacedName.String()]); err != nil {
+			r.c.Log.WithFields(logrus.Fields{
+				"entryid": r.spiffeIDCollection[req.NamespacedName.String()],
+			}).WithError(err).Error("unable to delete spire entry", )
+			return ctrl.Result{}, err
+		}
+		delete(r.spiffeIDCollection, req.NamespacedName.String())
+		r.c.Log.WithFields(logrus.Fields{
+			"entry": req.NamespacedName.String(),
+		}).Info("Finalized entry")
+
+		return ctrl.Result{}, client.IgnoreNotFound(err)
+	}
+
+	entryId, err := r.updateOrCreateSpiffeID(ctx, &spiffeID)
+	if err != nil {
+		r.c.Log.WithFields(logrus.Fields{
+			"request": req,
+		}).WithError(err).Error(err, "unable to get or create spire entry")
+		return ctrl.Result{}, err
+	}
+
+	oldEntryId := spiffeID.Status.EntryId
+	if oldEntryId == nil || *oldEntryId != entryId {
+		// We need to update the Status field
+		if oldEntryId != nil {
+			// entry resource must have been modified, delete the hanging one
+			if err := r.ensureDeleted(ctx, *spiffeID.Status.EntryId); err != nil {
+				r.c.Log.WithFields(logrus.Fields{
+					"entryid": *spiffeID.Status.EntryId,
+				}).WithError(err).Error("unable to delete old spire entry")
+				return ctrl.Result{}, err
+			}
+		}
+		spiffeID.Status.EntryId = &entryId
+		if err := r.Status().Update(ctx, &spiffeID); err != nil {
+			r.c.Log.WithError(err).Error("unable to update SpiffeID status")
```

Comment:
```suggestion
			r.c.Log.WithError(err).Error("Unable to update SpiffeID CRD status")
```

Reply:
Should probably be "resource" rather than "CRD." The CRD is the definition and schema for the resource, not the resource itself.